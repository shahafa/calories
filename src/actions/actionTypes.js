export const RESET_AUTH_STORE = 'RESET_AUTH_STORE';
export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const GET_MEALS_REQUEST = 'GET_MEALS_REQUEST';
export const GET_MEALS_SUCCESS = 'GET_MEALS_SUCCESS';
export const GET_MEALS_FAILURE = 'GET_MEALS_FAILURE';
export const ADD_MEAL_REQUEST = 'ADD_MEAL_REQUEST';
export const ADD_MEAL_SUCCESS = 'ADD_MEAL_SUCCESS';
export const ADD_MEAL_FAILURE = 'ADD_MEAL_FAILURE';
export const DELETE_MEAL_REQUEST = 'DELETE_MEAL_REQUEST';
export const DELETE_MEAL_SUCCESS = 'DELETE_MEAL_SUCCESS';
export const DELETE_MEAL_FAILURE = 'DELETE_MEAL_FAILURE';
export const EDIT_MEAL_REQUEST = 'EDIT_MEAL_REQUEST';
export const EDIT_MEAL_SUCCESS = 'EDIT_MEAL_SUCCESS';
export const EDIT_MEAL_FAILURE = 'EDIT_MEAL_FAILURE';
export const CLOSE_MEALS_ERROR_SNACKBAR = 'CLOSE_MEALS_ERROR_SNACKBAR';
export const SET_FILTER = 'SET_FILTER';

export const GET_USERS_REQUEST = 'GET_USERS_REQUEST';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_USERS_FAILURE = 'GET_USERS_FAILURE';
export const UPDATE_USERS_ROLE_REQUEST = 'UPDATE_USERS_ROLE_REQUEST';
export const UPDATE_USERS_ROLE_SUCCESS = 'UPDATE_USERS_ROLE_SUCCESS';
export const UPDATE_USERS_ROLE_FAILURE = 'UPDATE_USERS_ROLE_FAILURE';

export const GET_SETTINGS_REQUEST = 'GET_SETTINGS_REQUEST';
export const GET_SETTINGS_SUCCESS = 'GET_SETTINGS_SUCCESS';
export const GET_SETTINGS_FAILURE = 'GET_SETTINGS_FAILURE';
export const SET_SETTINGS_REQUEST = 'SET_SETTINGS_REQUEST';
export const SET_SETTINGS_SUCCESS = 'SET_SETTINGS_SUCCESS';
export const SET_SETTINGS_FAILURE = 'SET_SETTINGS_FAILURE';
export const CLOSE_SETTINGS_ERROR_SNACKBAR = 'CLOSE_SETTINGS_ERROR_SNACKBAR';
