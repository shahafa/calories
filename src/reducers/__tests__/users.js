import usersReducer from '../users';
import * as types from '../../constants/actionTypes';

describe('users reducer', () => {
  it('should return the initial state', () => {
    expect(
      usersReducer(undefined, {}),
    ).toMatchSnapshot();
  });

  it('should start get users request', () => {
    expect(
      usersReducer(undefined, {
        type: types.GET_USERS_REQUEST,
      }),
    ).toMatchSnapshot();
  });

  it('should set users after successful get users request', () => {
    expect(
      usersReducer(undefined, {
        type: types.GET_USERS_SUCCESS,
        payload: {
          data: {
            users: [{
              id: '9e65644e-e326-48d9-be7d-7d13f8b36992',
              email: 'user@mail.com',
              role: 'user',
            }],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should set users loading flag to false after get users failure', () => {
    expect(
      usersReducer(undefined, {
        type: types.GET_USERS_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should update user role after successful update user role request', () => {
    expect(
      usersReducer(undefined, {
        type: types.UPDATE_USERS_ROLE_SUCCESS,
        payload: {
          data: {
            users: [{
              id: '9e65644e-e326-48d9-be7d-7d13f8b36992',
              email: 'user@mail.com',
              role: 'userManager',
            }],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should revert users role after update user role failure', () => {
    expect(
      usersReducer(undefined, {
        type: types.UPDATE_USERS_ROLE_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should delete user locally before delete user request', () => {
    expect(
      usersReducer(undefined, {
        type: types.DELETE_USER_REQUEST,
        meta: {
          userId: '9e65644e-e326-48d9-be7d-7d13f8b36992',
        },
      }),
    ).toMatchSnapshot();
  });

  it('should update users list after successful delete user request', () => {
    expect(
      usersReducer(undefined, {
        type: types.DELETE_USER_SUCCESS,
        payload: {
          data: {
            users: [{
              id: '9e65644e-e326-48d9-be7d-7d13f8b36992',
              email: 'user@mail.com',
              role: 'user',
            }],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should revert users list after delete user failure', () => {
    expect(
      usersReducer(undefined, {
        type: types.DELETE_USER_FAILURE,
      }),
    ).toMatchSnapshot();
  });
});
