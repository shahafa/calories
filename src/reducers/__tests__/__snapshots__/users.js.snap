// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users reducer should delete user locally before delete user request 1`] = `
Object {
  "isLoading": false,
  "users": Array [],
}
`;

exports[`users reducer should return the initial state 1`] = `
Object {
  "isLoading": false,
  "users": Array [],
}
`;

exports[`users reducer should revert users list after delete user failure 1`] = `
Object {
  "isLoading": false,
  "users": Array [
    Object {
      "email": "user@mail.com",
      "id": "9e65644e-e326-48d9-be7d-7d13f8b36992",
      "role": "user",
    },
  ],
}
`;

exports[`users reducer should revert users role after update user role failure 1`] = `
Object {
  "isLoading": false,
  "users": Array [
    Object {
      "email": "user@mail.com",
      "id": "9e65644e-e326-48d9-be7d-7d13f8b36992",
      "role": "userManager",
    },
  ],
}
`;

exports[`users reducer should set users after successful get users request 1`] = `
Object {
  "isLoading": false,
  "users": Array [
    Object {
      "email": "user@mail.com",
      "id": "9e65644e-e326-48d9-be7d-7d13f8b36992",
      "role": "user",
    },
  ],
}
`;

exports[`users reducer should set users loading flag to false after get users failure 1`] = `
Object {
  "isLoading": false,
  "users": Array [],
}
`;

exports[`users reducer should start get users request 1`] = `
Object {
  "isLoading": true,
  "users": Array [],
}
`;

exports[`users reducer should update user role after successful update user role request 1`] = `
Object {
  "isLoading": false,
  "users": Array [
    Object {
      "email": "user@mail.com",
      "id": "9e65644e-e326-48d9-be7d-7d13f8b36992",
      "role": "userManager",
    },
  ],
}
`;

exports[`users reducer should update users list after successful delete user request 1`] = `
Object {
  "isLoading": false,
  "users": Array [
    Object {
      "email": "user@mail.com",
      "id": "9e65644e-e326-48d9-be7d-7d13f8b36992",
      "role": "user",
    },
  ],
}
`;
